eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java

```diff
@@ -187,6 +199,39 @@ private Collection<Dependency> deps() throws IOException {
             .collect(Collectors.toList());
     }
 
+    /**
+     * Check dependencies for conflicts.
+     * @param deps Dependencies
+     * @param fail Fail on conflicts or just warn
+     */
+    private static void checkConflicts(
+        final Collection<Dependency> deps,
+        final boolean fail
+    ) {
+        final Map<String, Set<String>> grouped = deps.stream()
+            .collect(
+                Collectors.groupingBy(
+                    Dependency::getManagementKey,
+                    Collectors.mapping(
+                        Dependency::getVersion,
+                        Collectors.toSet()
+                    )
+                )
+            );
+        for (final Map.Entry<String, Set<String>> vers : grouped.entrySet()) {
+            if (vers.getValue().size() > 1) {
+                final String msg = String.format(
+                    "Conflicting dependencies are found for %s: %s",
+                    vers.getKey(), vers.getValue()
+                );
+                if (fail) {
```

Comment:
@mximp this algorithm will fail on the _first_ conflict, while all others won't be visible to the user. Maybe it's better to print them into the log and only then throw an exception, after all logging messages?

Reply:
Agree.. good idea.